core/src/main/java/org/infinispan/functional/impl/MetaParamsExternalizer.java

```diff
@@ -0,0 +1,39 @@
+package org.infinispan.functional.impl;
+
+import java.io.IOException;
+import java.io.ObjectInput;
+import java.io.ObjectOutput;
+import java.util.Set;
+
+import org.infinispan.commons.api.functional.MetaParam;
+import org.infinispan.commons.marshall.AbstractExternalizer;
+import org.infinispan.commons.util.Util;
+import org.infinispan.marshall.core.Ids;
+
+public final class MetaParamsExternalizer extends AbstractExternalizer<MetaParams> {
```

Comment:
Makes sense. `marshall`is trivial - but should `unmarshall` return an iterable? Or just use `unmarshallArray` and make sure that these two are compatible?